src/object/srv_ec_aggregate.c

```diff
@@ -649,17 +1642,56 @@ agg_process_stripe(struct ec_agg_entry *entry)
 		update_vos = false;
 		goto out;
 	}
+	/* Parity, some later replicas, possibly holes, not full stripe. */
+	if (entry->ae_cur_stripe.as_has_holes)
+		process_holes = true;
+	else
+		rc = agg_process_partial_stripe(entry);
+
 out:
-	if (update_vos && rc == 0) {
-		entry->ae_cur_stripe.as_suffix_ext = agg_get_carry_under(entry);
-		rc = agg_update_vos(entry, write_parity);
+	if (process_holes && rc == 0)
+		rc = agg_process_holes(entry);
```

Comment:
this seems can be moved above, under if (has_holes), so you do not need process_holes?  and also more consistent with other processing?